.github/workflows/unit-tests.yml

```diff
@@ -184,24 +184,26 @@ jobs:
           # matrix.mode = high-deps
           echo "$COMPONENTS" | xargs -n1 | parallel -j +3 "_run_tests {} 'cd {} && $COMPOSER_UP && $PHPUNIT$LEGACY'" || X=1
 
+          # get a list of the patched components (relies on .github/build-packages.php being called in the previous step)
           (cd src/Symfony/Component/HttpFoundation; mv composer.bak composer.json)
-          COMPONENTS=$(git diff --name-only src/ | grep composer.json || true)
+          PATCHED_COMPONENTS=$(git diff --name-only src/ | grep composer.json || true)
 
-          if [[ $COMPONENTS && $SYMFONY_VERSION = *.4 ]]; then
+          # for the x.4 branch, checkout and test previous major with the patched components (only for patched components)
```

Comment:
for x.4 branches (there can be many, eg 4.4 & 5.4)

Reply:
fixed